packages/jmix-react-antd/src/ui/table/DataTable.tsx

```diff
@@ -417,6 +441,37 @@ class DataTableComponent<
     }
   };
 
+  renderBodyOnlyVisible = (props: any) => {
+    if (!Array.isArray(props.children)) {
+      return props.children;
+    }
+
+    return <tr className={props.className}>
+      {props.children.filter((col: any) => this.fieldsVisibility.get(col.key as string))}
+    </tr>
+  }
+  renderHeaderOnlyVisible = (props: any) => {
+    if (!Array.isArray(props.children)) {
+      return props.children;
+    }
+
+    return (
+      <tr>
+        {props.children.filter((col: any) => this.fieldsVisibility.get(col.key as string))}
+      </tr>
+    );
+  }
+
+  changeFieldVisibility = (key: string, value: boolean) => {
+    const newFieldsVisibility = new Map(this.fieldsVisibility);
+    newFieldsVisibility?.set(key, value);
+    this.fieldsVisibility = newFieldsVisibility;
```

Comment:
Do we really need to recreate the map? I think that since we have [proxies](https://mobx.js.org/configuration.html#proxy-support) enabled (default in MobX 6), adding/deleting entries should be picked up automatically.

**If it doesn't**: does this approach has an advantage over using `observable.map`?